### PR TITLE
feat: docs generator v2

### PR DESCRIPTION
**Pull request by saravmajestic**
## Overview

- includes common react webview panel which can be extended for other panels. Details on new web view is [here](https://github.com/AltimateAI/vscode-dbt-power-user/tree/feat/docs-v2/webview_panels)
- includes docs gen changes
- includes datapilot side panel 

### How to test:
- Go to documentation editor tab and click `show new ux` button to load new UI

## Checklist

- [ ] I have run this code and it appears to resolve the stated issue
- [ ] `README.md` updated and added information about my change

### CODE REVIEW

File: src/manifest/dbtProjectContainer.ts [hunk 1]
```diff
@@ -184,6 +184,10 @@ export class DBTProjectContainer implements Disposable {
     return this.context!.extensionUri;
   }
 
+  get extensionPath() {
+    return this.context!.extensionPath;
+  }
+
```
Comment (mdesmet) [inline src/manifest/dbtProjectContainer.ts:190]: In other places we use extensionUri, why we need path? Will it work on windows?

File: src/webview_provider/altimateWebviewProvider.ts [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+  }
+
+  private setupWebviewOptions(context: WebviewViewResolveContext) {
+    this._panel!.description = this.panelDescription;
+    this._panel!.webview.options = <WebviewOptions>{
+      enableScripts: true,
+      localResourceRoots: [
+        Uri.file(
+          path.join(
+            this.dbtProjectContainer.extensionPath,
```
Comment (mdesmet) [inline src/webview_provider/altimateWebviewProvider.ts:176]: Fishy, should use uri.fesPath or won't work on windows.

File: src/webview_provider/altimateWebviewProvider.ts [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+          await commands.executeCommand("dbtPowerUser.datapilot-webview.focus");
+          // Adding timeout to let the datapilot complete rendering
+          setTimeout(() => {
+            sharedStateManager.postMessage(message);
+          }, 500);
+          break;
+        case "updateConfig":
+          console.log(
+            "Updating config",
+            (params as UpdateConfigProps).key,
```
Comment (mdesmet) [inline src/webview_provider/altimateWebviewProvider.ts:104]: extract params to typed variable.

File: src/webview_provider/types.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+export type UpdateConfigProps = {
+  key: string;
+  value: string | boolean | number;
+  isPreviewFeature?: boolean;
+};
```
Comment (mdesmet) [inline src/webview_provider/types.ts:5]: If only used in alitmateWebViewProvider. keep it only there and don't export.

File: src/webview_provider/sharedStateManager.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  addListener(listener: (message: any) => void) {
+    this.listeners.push(listener);
+  }
+
+  postMessage(message: any) {
+    this.listeners.forEach((listener) => listener(message));
+  }
+}
+
+export const sharedStateManager = SharedStateManager.getInstance();
```
Comment (mdesmet) [inline src/webview_provider/sharedStateManager.ts:23]: Why not use vscode event emitters?

File: src/webview_provider/altimateWebviewProvider.ts (hunk 2 above)
Comment (saravmajestic) [reply src/webview_provider/altimateWebviewProvider.ts:176]: ok, i ll try with uri.fsPath

File: src/webview_provider/types.ts (hunk 4 above)
Comment (saravmajestic) [reply src/webview_provider/types.ts:5]: This is used in datapilot panel as well and will be used in other panels in future if needed

File: src/webview_provider/sharedStateManager.ts (hunk 5 above)
Comment (saravmajestic) [reply src/webview_provider/sharedStateManager.ts:23]: I will try that

File: src/webview_provider/types.ts (hunk 4 above)
Comment (mdesmet) [reply src/webview_provider/types.ts:5]: but if it is part of the panel it is inherited and will be present on the object that is using it. So I don't immediately see the need to export it and create another file.

File: src/webview_provider/altimateWebviewProvider.ts [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+              "dbtPowerUser.datapilot-webview.focus",
+            );
+          }
+          break;
+        case "datapilot:message":
+          await commands.executeCommand("dbtPowerUser.datapilot-webview.focus");
+          // Adding timeout to let the datapilot complete rendering
+          setTimeout(() => {
+            sharedStateManager.postMessage(message);
+          }, 500);
```
[the remark this answers is not in the document]
Comment (mdesmet) [reply src/webview_provider/altimateWebviewProvider.ts:99]: The right way is to let the panel send a message to let it tell you the panel is ready to receive messages.

File: src/webview_provider/sharedStateManager.ts (hunk 5 above)
Comment (saravmajestic) [reply src/webview_provider/sharedStateManager.ts:23]: I can build event emitters in 2 ways:
- define a static variable in `AltimateWebviewProvider` and listen to the events in sub classes 
- define a service for event emitter, inject it in AltimateWebviewProvider and sub classes and listen to the events

It is required to use same EventEmitter instance in all classes to listen to the fired events.

I prefer to use static variable as eventemitter will not be mutated and keeps it simple. If we go with service, then need to inject in all classes and use it. Lmk your thoughts @mdesmet

File: src/webview_provider/altimateWebviewProvider.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (saravmajestic) [reply src/webview_provider/altimateWebviewProvider.ts]: @mdesmet here is the [commit](https://github.com/AltimateAI/vscode-dbt-power-user/pull/801/commits/de319de462ec0e99432144fa68a1d69760e54024) to address eventemitter and webview ready handler

File: src/webview_provider/sharedStateManager.ts (hunk 5 above)
Comment (mdesmet) [reply src/webview_provider/sharedStateManager.ts]: I think you just need to extract your eventmitter in one class and inject it as a singleton through inversify.

**Review by mdesmet — APPROVED** [review]

**Pull request merged**